datasetinsights/data/datasets/groceries_real.py

```diff
@@ -157,6 +176,92 @@ def _filepath(self, filename):
 
+    @staticmethod
+    def _download_http(source_uri, dest_path, version):
+        """ Download dataset from Public HTTP URL.
+
+        Args:
+            source_uri (str): source url where the file should be downloaded
+            dest_path (str): destination path of the file
+
+        Raises:
+            DownloadError if the download file failed
+            ChecksumError if the download file checksum does not match
+        """
+
+        try:
+            logger.info("Downloading the dataset.")
+            download_file(source_uri=source_uri, dest_path=dest_path)
+        except DownloadError as e:
+            logger.info(
+                f"The request download from {source_uri} -> {dest_path} can't "
+                f"be completed."
+            )
+            raise e
+        expected_checksum = GroceriesReal.GROCERIES_REAL_DATASET_TABLES[
+            version
+        ].checksum
+        try:
+            validate_checksum(dest_path, expected_checksum)
+        except ChecksumError as e:
+            logger.info("Checksum mismatch. Delete the downloaded files.")
+            os.remove(dest_path)
+            raise e
+
+    @staticmethod
+    def _extract_file(dest_path, root_dir):
+        """ Unzip the downloaded file.
+        """
+        logger.info("Unzipping the dataset file.")
+        with zipfile.ZipFile(dest_path, "r") as zip_dir:
+            zip_dir.extractall(root_dir)
+
+    @staticmethod
+    def download(data_root, version):
+        """ Download dataset from Public HTTP URL.
+
+        If the file already exists and the checksum matches, it will skip the
+        download step. If not, it would delete the previous file and download
+        it again. If the file doesn't exist, it would download the file.
+
+        Args:
+            data_root (str): Root directory prefix of datasets
+            version (str): version of GroceriesReal dataset, e.g. "v3"
+
+        Raises:
+            ValueError if the dataset version is not supported
+            ChecksumError if the download file checksum does not match
+            DownloadError if the download file failed
+        """
+        if version not in GroceriesReal.GROCERIES_REAL_DATASET_TABLES.keys():
+            raise ValueError(
+                f"A valid dataset version is required. Available versions are:"
+                f"{GroceriesReal.GROCERIES_REAL_DATASET_TABLES.keys()}"
+            )
+        dest_path = os.path.join(
+            data_root, GroceriesReal.LOCAL_PATH, f"{version}.zip"
+        )
+        expected_checksum = GroceriesReal.GROCERIES_REAL_DATASET_TABLES[
+            version
+        ].checksum
+        extract_folder = os.path.join(data_root, GroceriesReal.LOCAL_PATH)
+        if os.path.exists(dest_path):
+            logger.info("The dataset file exists. Skip download.")
+            try:
+                validate_checksum(dest_path, expected_checksum)
+            except ChecksumError:
+                logger.info(
```

Comment:
[nit] Since this hides ChecksumError from execution, maybe it's also worth to log error message as warning using `logger.warning(ex)`